corehq/apps/commtrack/views.py

```diff
@@ -168,42 +169,26 @@ def post(self, request, *args, **kwargs):
 @login_and_domain_required
 def api_query_supply_point(request, domain):
     id = request.GET.get('id')
-    query = request.GET.get('name', '')
-    
+    query = request.GET.get('name', '').lower()
+
     def loc_to_payload(loc):
-        return {'id': loc._id, 'name': loc.name}
+        return {'id': loc.location_id, 'name': loc.display_name}
 
     if id:
         try:
-            loc = Location.get(id)
-            return HttpResponse(json.dumps(loc_to_payload(loc)), 'text/json')
-
-        except ResourceNotFound:
-            return HttpResponseNotFound(json.dumps({'message': 'no location with id %s found' % id}, 'text/json'))
-
+            loc = SQLLocation.objects.get(location_id=id)
+        except SQLLocation.DoesNotExist:
+            return json_response(
+                {'message': 'no location with id %s found' % id},
+                status_code=404,
+            )
+        else:
+            return json_response(loc_to_payload(loc))
     else:
```

Comment:
I switched this to use SQLLocation.  Now it supports infix lookups too ("ass" will match "Massachusetts")